cli/test/smokehouse/smokehouse.js

```diff
@@ -195,6 +195,13 @@ async function runSmokeTest(smokeTestDefn, testOptions) {
         ...await lighthouseRunner(requestedUrl, configJson, {isDebug, useLegacyNavigation}),
         networkRequests: takeNetworkRequestUrls ? takeNetworkRequestUrls() : undefined,
       };
+
+      if (!result || !result.lhr?.audits || !result.artifacts) {
+        // Don't try again–something went really wrong! ex: the config could not be resolved
+        i = retries + 1;
+        throw new Error('lighthouse runner returned a bad result. got lhr:\n' +
+          JSON.stringify(result.lhr, null, 2));
+      }
```

Comment:
"just remove all the bugs" is a fair point :)

What about aiming for runner-specific checking in the runner, but keep a generic check here? e.g.

```suggestion
      if (!result.lhr?.audits || !result.artifacts) {
        // Something went really wrong and the runner didn't catch it.
        throw new Error('lighthouse runner returned a bad result. got lhr:\n' +
          JSON.stringify(result.lhr, null, 2));
      }
```

Skipping the retries does seem unnecessary (especially by manipulating `i` :P); if it's that bad and it's happening outside of debugging we really do need to fix the runner